packages/cms/lib/modules/resource-form-widgets/lib/api.js

```diff
@@ -0,0 +1,180 @@
+module.exports = async function(self, options) {
+  /**
+   * Sync data with api when user saves or commit
+   * @param req
+   * @param doc
+   * @param options
+   * @returns {Promise<void>}
+   */
+  self.syncConfirmationFields = async (req, doc, options) => {
+    if ((doc.workflowLocale && doc.workflowLocale === 'default-draft') || (!doc.body || !doc.body.items)) {
+      return;
+    }
+
+    doc.body.items.forEach((item) => {
+      const items = self.getAreaItems(item);
+
+      items.forEach(async (item) => {
+        if (item.type === 'resource-form') {
+          if (item.confirmationEnabledUser) {
+            try {
+              await self.addOrUpdateNotification(item, 'User');
+            } catch (error) {
+              console.error(
+                'something went wrong when update admin confirmation settings to api',
+                error.message
+              );
+            }
+          } else {
+            await self.disableNotificationRuleSet(`User-${item.formName}`);
+          }
+
+          if (item.confirmationEnabledAdmin) {
+            try {
+              await self.addOrUpdateNotification(item, 'Admin');
+            } catch (error) {
+              console.error(
+                'something went wrong when update admin confirmation settings to api',
+                error.message
+              );
+            }
+          } else {
+            await self.disableNotificationRuleSet(`Admin-${item.formName}`);
+          }
+        }
+      })
+    })
+  }
+
+  /**
+   * Add or update confirmation data to the api
+   * @param {object} item
+   * @param {string} type 'User' or 'Admin'
+   * @returns {Promise<void>}
+   */
+  self.addOrUpdateNotification = async (item, type) => {
+    const formName = `${type}-${item.formName}`;
+
+    const { ruleset, template, recipient } = await self.getNotificationByFormName(formName);
+
+    const templateData = {
+      label: formName,
+      subject: item[`confirmationSubject${type}`],
+      text: item[`confirmationEmailContent${type}`],
+      templateFile: item[`confirmationTemplateName${type}`]
+    }
+
+    if (template) {
+      templateData.id = template.id
+    }
+
+    const updatedTemplate = await self.apos.openstadApi.addOrUpdateNotificationTemplate(templateData);
+    const ruleSetData = {
+      notificationTemplateId: updatedTemplate.id,
+      active: 1,
+      label: formName,
+      body: JSON.stringify({"and":[{"===":[{"var":"eventType"},"CREATE"]},{"===":[{"var":"resource"},item.resource]},{"===":[{"var":"instance.formName"}, item.formName]}]})
+    }
+    if (ruleset) {
+      ruleSetData.id = ruleset.id
+    }
+
+    const updatedRuleSet = await self.apos.openstadApi.addOrUpdateNotificationRuleSet(ruleSetData);
+    const recipientData = {
```

Comment:
```suggestion
    const updatedRuleSet = await self.apos.openstadApi.addOrUpdateNotificationRuleSet(ruleSetData);
    
    const recipientData = {
```

Zou hier ook iets witruimte toevoegen